src/rpdk/core/contract/resource_client.py

```diff
@@ -288,22 +288,36 @@ def assert_failed(status, response):
         return error_code
 
     @staticmethod
+    # pylint: disable=R0913
     def make_request(
         desired_resource_state,
         previous_resource_state,
         region,
         account,
         partition,
+        action,
+        creds,
+        client_token,
+        callback_context=None,
         **kwargs
     ):
         return {
-            "desiredResourceState": desired_resource_state,
-            "previousResourceState": previous_resource_state,
-            "logicalResourceIdentifier": None,
-            **kwargs,
+            "requestData": {
+                "callerCredentials": creds,
+                "providerCredentials": creds,
```

Comment:
How does this behave when we use execution role for contract tests? It might be better to set these to null along with provider log group name: https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/blob/master/src/main/java/software/amazon/cloudformation/LambdaWrapper.java#L151-L155. The caller credentials might not have permission to create logs or post logs.

Reply:
I am unsure how we would run contract test with these 2 parameters set as null. I ran it locally and this fails.

Reply:
right now this is default behavior. There is a issue which I need to address next about this issue: https://github.com/aws-cloudformation/cloudformation-cli/issues/544

Reply:
yep, the `providerCredentials` should set to `null`. Otherwise, the contract test will pollute the provider metrics.  With null, it won't publish or log anything.